scripts/install.sh

```diff
@@ -1,18 +1,93 @@
 #!/bin/bash
 
-# user permission check
+# User Permission Check
 if [ ! $(id -u) = 0 ]; then
   echo "Error: GlobaLeaks install script must be run by root"
   exit 1
 fi
 
+# Preliminary Requirements Check
+ERR=0
+echo "Checking preliminary GlobaLeaks requirements"
+for REQ in apt-key apt-get gpg curl
+do
+  if which $REQ >/dev/null; then
+    echo " + $REQ requirement meet"
+  else
+    ERR=$(($ERR+1))
+    echo " - $REQ requirement not meet"
+  fi
+done
+
+if [ $ERR -ne 0 ]; then
+  echo "Error: Found ${ERR} unmet requirements"
+  exit 1
+fi
+
+DO () {
+  if [ -z "$2" ]; then
+    EXPECTED_RET=0
+  else
+    EXPECTED_RET=$2
+  fi
+  if [ -z "$3" ]; then
+    CMD=$1
+  else
+    CMD=$3
+  fi
+  echo -n "Running: \"$CMD\"... "
+  eval $CMD &>${LOGFILE}
+
+  STATUS=$?
+
+  echo $CMD > $TMPDIR/last_command
+  echo $STATUS > $TMPDIR/last_status
+
+  if [ "$STATUS" -eq "$EXPECTED_RET" ]; then
+    echo "SUCCESS"
+  else
+    echo "FAIL"
+    echo "COMBINED STDOUT/STDERR OUTPUT OF FAILED COMMAND:"
+    cat ${LOGFILE}
+    echo "Please help us diagnose the failure and report the installation error on https://forum.globaleaks.org"
```

Comment:
I suggest the comment to be.

> Ouch! The installation failed. Please help us out and report the failure on the forum so that others don't have the same issue. https://forum.globaleaks.org